prime-router/src/main/kotlin/Mappers.kt

```diff
@@ -80,6 +80,8 @@ class MiddleInitialMapper : Mapper {
             return null
         } else {
             if (values.size != 1) error("Found ${values.size} values.  Expecting 1 value. Args: $args, Values: $values")
+            if (values.first().value.isEmpty())
+                return null
             return values.first().value.substring(0..0).toUpperCase()
```

Comment:
This line expects a string of length 1.  I had a case where there was a non-null string, but it was empty, which causes an exception here.

Reply:
Actually, just googling substring in Kotlin, I think the above always creates an empty string.

Has this MiddleEarthMapper ever worked??